### DISCUSSION TITLE
Is it possible to get enum name from `meta::uvalue`?

### DISCUSSION THREAD

**Discussion by MineBill** [BlackMATov/meta.hpp#93] [Q&A] [answered]
I'm using something like this to draw a dropdown with the enum value as options:
```cpp
for (auto const& evalue : enum_type.get_evalues()) {
    if (ImGui::Selectable(evalue.get_name().c_str())) {
        member.set(ptr, evalue.get_value());
    }
}
```
It works fine and i can set the member just fine. But I can't seem to find a way to go from a `meta::uvalue` (which is what `member.get(ptr)` returns) to a `meta::evalue` so i can then get the name of the enumeration. Is that possible?

**Comment by BlackMATov** [comment]
Hm... It's a good feature request, because right now we can't get an `evalue` from dynamic values (only from static typed values). I'm going to refactor `enum_type` in the next few days!

**Comment by MineBill** [comment]
Thanks for replying! I appreciate your help!
I figured out a whacky way to do this that i want to post just in case it helps someone in the future in any way:
```cpp
auto GetName = [&enum_type](int i) -> std::string_view {
    for (auto const& evalue : enum_type.get_evalues()) {
        if (auto as_underlying_ptr = static_cast<int const*>(evalue.get_underlying_value().get_data()); *as_underlying_ptr == i) {
            return evalue.get_name();
        }
    }
    return std::string_view{ "Invalid" };
};

auto as_underlying_ptr = static_cast<int**>(value.get_data());
auto name = GetName(**as_underlying_ptr);
```

I guess this only really works because i don't specify an underlying type for my enums so it's `int` be default.
Also, i'm using double pointers here because the members are registered with the `as_pointer` policy because i need to modify them.

In any case, i'm waiting for the refactor!

**Comment by BlackMATov** [comment, marked as answer]
So, `enum_type` was refactored. Now it contains `value_to_evalue` overloaded methods that work with static and dynamic values as well, and can be used in your case!

Here is some examples/tests of this new functionality:
https://github.com/BlackMATov/meta.hpp/blob/main/develop/untests/meta_issues/github_discussion_93.cpp
